rust/bridge/node/src/lib.rs

```diff
@@ -3,13 +3,289 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+use async_trait::async_trait;
+use libsignal_bridge::node;
+use libsignal_protocol::*;
+use neon::context::Context;
 use neon::prelude::*;
+use signal_neon_futures::*;
+use std::fmt;
+use std::marker::PhantomData;
+use std::panic::AssertUnwindSafe;
+use std::sync::Arc;
 
 pub mod logging;
 
+#[derive(Debug)]
+struct CallbackError {
```

Comment:
This is what signal_neon_futures::PersistentException is for, though admittedly it may need some wrapping or extra traits to really work. (Having to explicitly destroy it might make that unworkable too.)